src/app/web-request-wrapper.tsx

```diff
@@ -0,0 +1,38 @@
+import { IS_TEST_ENV } from '@shared/constants';
+import { useHasAllowedDiagnostics } from './store/onboarding/onboarding.hooks';
+
+type Props = {};
+
+let tabOrigin = ''; // global, per content-script
+
+chrome.tabs.query({ active: true, lastFocusedWindow: true }, tabs => {
```

Comment:
Curious what's going on here: So, on each frame creation, we query the active tab (either full page and the extension tab, or popup mode, and a web page), cache the origin, and on subsequent requests, if the initiator is the extension, add the referrer? 🤔